webapp/channels/src/components/team_sidebar/components/team_button.tsx

```diff
@@ -36,186 +30,215 @@ interface Props {
     placement?: 'left' | 'right' | 'top' | 'bottom';
     teamIconUrl?: string | null;
     switchTeam: (url: string) => void;
-    intl: IntlShape;
     isDraggable?: boolean;
     teamIndex?: number;
     teamId?: string;
     isInProduct?: boolean;
     hasUrgent?: boolean;
 }
 
-class TeamButton extends React.PureComponent<Props> {
-    handleSwitch = (e: React.MouseEvent) => {
+export default function TeamButton({
+    btnClass,
+    url,
+    displayName,
+    order,
+    unread,
+    mentions,
+    teamIconUrl,
+    isDraggable = false,
+    switchTeam,
+    teamIndex,
+    teamId,
+    tip,
+    ...props
+}: Props) {
+    const {formatMessage} = useIntl();
+
+    const handleSwitch = useCallback((e: React.MouseEvent) => {
         mark('TeamLink#click');
         e.preventDefault();
-        this.props.switchTeam(this.props.url);
+        switchTeam(url);
 
         setTimeout(() => {
             trackEvent('ui', 'ui_team_sidebar_switch_team');
         }, 0);
-    };
+    }, [switchTeam, url]);
 
-    render() {
-        const {teamIconUrl, displayName, btnClass, mentions, unread, isDraggable = false, teamIndex, teamId, order} = this.props;
-        const {formatMessage} = this.props.intl;
+    let teamClass: string = props.active ? 'active' : '';
+    const isNotCreateTeamButton: boolean = !url.endsWith('create_team') && !url.endsWith('select_team');
 
-        let teamClass: string = this.props.active ? 'active' : '';
-        const isNotCreateTeamButton: boolean = !this.props.url.endsWith('create_team') && !this.props.url.endsWith('select_team');
+    let badge: JSX.Element | undefined;
 
-        let badge: JSX.Element | undefined;
+    let ariaLabel = formatMessage({
```

Comment:
Worst case scenario, we are calling formatMessage 3 times. It is not expensive, but it is also not free. It is out of the scope of this PR, but we could re-consider how to make this badge/teamClass/ariaLabel logic more legible.

Reply:
That is weird that it's written that way. I feel like whoever wrote this didn't like if-else blocks, but I'm also going to leave it as-is since it's mostly just weird looking even if there is a tiny performance improvement here